assets/eip-3525/contracts/ERC3525Upgradeable.sol

```diff
@@ -0,0 +1,718 @@
+//SPDX-License-Identifier: CC0-1.0
+pragma solidity ^0.8.0;
+
+import "./IERC3525.sol";
+import "./IERC3525Receiver.sol";
+import "./extensions/IERC3525Metadata.sol";
+import "./openzeppelin/ContextUpgradeable.sol";
+import "./openzeppelin/ERC165Upgradeable.sol";
+import "./openzeppelin/IERC721Enumerable.sol";
+import "./openzeppelin/IERC721Metadata.sol";
+import "./openzeppelin/IERC721Receiver.sol";
+import "./utils/base64.sol";
+import "./utils/StringConvertor.sol";
+import "hardhat/console.sol";
```

Comment:
This feels like it is out of scope for a reference implementation.  The reference implementation should do the bare minimum required to show how one might create a very simple implementation of this standard, and since this isn't an upgradability standard this whole file is out of scope and adds unnecessary complexity to the reference implementation.